[WIP] Add unit tests

HubLot opened the pull request:
Hi,

I'm currently adding unit tests into the package to increase its 'robustness', specially on the different functions of the package (I think the scripts in `test_regression.py` are very well covered).
I also realized that I introduced some functions before without unit tests associated (booo).

HubLot commented:
Hi,

Here a few commits to close this PR:
 - I shrinked the `test_data` directory (~20Mb -> 2Mb) by shrinking the barstar test files. I removed the water from the gro and xtc and kept only 100 frames of the xtc. I also kept only 3 models of the 2LFU pdb (instead of 10).

 - As discussed in #121, I created a function called `tests()` within the root path of PBxplore so the users can run the tests easily. For example :
```python
import pbxplore
pbxplore.tests()
```
One small issue: usually the test folder is named "tests" and the function is named "test". Here, it's the other way. I could rename those if you want before merging.
Another one is that nose automatically find tests by researching keywords. To prevent it to run this new function `tests()`, we need to specify the test folder now:
```bash
nosetests -v pbxplore/test
```
I updated those information into the install page of the documentation by creating a chapter called "Testing PBxplore".

- I create a function named `system_info` inside the test module which print the version and the install directory of PBxplore dependencies as well as the version of python.

pierrepo commented:
Great work @HubLot !

> One small issue: usually the test folder is named "tests" and the function is named "test". Here, it's the other way. I could rename those if you want before merging.

If you feel this is more mainstream to have the test function named `test()` and the test directory `tests`, you can rename them.

Regarding `system_info` and the `module_info()` function, be aware that not all Python modules have their version number in `__version__`.

@jbarnoud what do you think?

HubLot commented:
> Regarding system_info and the module_info() function, be aware that not all Python modules have their version number in __version__.

Indeed, I'll add a test to prevent the possible error.

HubLot commented:
I updated the PR.

@jbarnoud  can you review?

jbarnoud commented:
I'll look at it on Saturday.

On 12-05-16 16:51, Hub wrote:
>
> I updated the PR.
>
> @jbarnoud <https://github.com/jbarnoud> can you review?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/pierrepo/PBxplore/pull/122#issuecomment-218781792>
>

jbarnoud commented on doc/source/installation.rst:
```diff
@@ -53,6 +53,29 @@ See the documentation of `pip <https://pip.pypa.io/en/stable/>`_ for more inform
+    $ python setup.py nosetests
+
+
+Or within the `PBxplore` package:
```
Does that mean that you must be in the PBxplore directory to tun this command? If not, the sentence is unclear. I would suggest something like:

    Or from the `PBxplore` module

jbarnoud commented on doc/source/installation.rst:
```diff
@@ -53,6 +53,29 @@ See the documentation of `pip <https://pip.pypa.io/en/stable/>`_ for more inform
+----------------
+
+`PBxplore` comes with unit tests and regression tests. It requires the package
+`nose <https://nose.readthedocs.io/en/latest/>`_ to run them. You have several options:
```
You have several options *to run the tests*:

jbarnoud commented on pbxplore/test/test_regression.py:
```diff
@@ -240,15 +251,10 @@ def test_xtc_input(self):
-                                    os.path.join(out_run_dir, output_fname))
-        else:
-            # MDanalysis is not available, PBassign should fail
-            assert status != 0, 'PBassign shoud not have exited with a 0 code'
```
The new tests does not make sure that PBxplore behave as expected if MDAnalysis is missing.

HubLot replied:
Indeed. I will reintroduce this one.
One can argue we have to do the same for matplotlib & weblogo

jbarnoud replied:
On 18/05/16 15:15, Hub wrote:
>
> In pbxplore/test/test_regression.py 
> <https://github.com/pierrepo/PBxplore/pull/122#discussion_r63699546>:
>
> > @@ -240,15 +251,10 @@ def test_xtc_input(self):
> >          status = exe.wait()
> >          print(out.decode('utf-8'))
> >          print(err.decode('utf-8'))
> > -        if IS_MDANALYSIS:
> > -            # MDanalysis is available, PBassign should run and produce the
> > -            # correct output
> > -            assert status == 0, 'PBassign exited with an error'
> > -            _assert_identical_files(os.path.join(REFDIR, output_fname),
> > -                                    os.path.join(out_run_dir, output_fname))
> > -        else:
> > -            # MDanalysis is not available, PBassign should fail
> > -            assert status != 0, 'PBassign shoud not have exited with a 0 code'
>
> Indeed. I will reintroduce this one.
> One can argue we have to do the same for matplotlib & weblogo
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/pierrepo/PBxplore/pull/122/files/7d5d91e364c3a1b35edaa7e65b56a0b6df186b71#r63699546>
>
I did not dear increase the load ;)

jbarnoud commented:
I posted a few comments. Besides these, everything looks good to me.

HubLot commented:
I updated the PR with @jbarnoud comments.
I also renamed the test folder "tests" and the function 'tests()' into 'test()' to be more 'mainstream' (and of course, updated the documentation and travis).
If you don't agree, I can roll back to the previous version.
If you agree, the PR can be merged :)
